dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java

```diff
@@ -151,11 +152,15 @@ public void setPath(String path) {
     }
 
     public String getAbsolutePath() {
-        return absolutePath;
+        if (absolutePath == null) {
+            return null;
+        }
+        return Paths.get(absolutePath).normalize().toString();
     }
 
     public void setAbsolutePath(String absolutePath) {
         // TODO: Figure out the actual absolute path before this workaround
+        // FIXME: it looks like dockstore tool test_parameter --add and a number of other CLI commands depend on this now
         this.absolutePath = ZipSourceFileHelper.addLeadingSlashIfNecessary((absolutePath));
```

Comment:
You could replace this with `StringUtils.prependIfMissing`, which I was unaware of before writing the `addLeadingSlashIfNecessary` method. But that might increase the scope of the change too much.